ludwig/models/llm.py

```diff
@@ -71,13 +71,27 @@ def update(self, modules: Dict[str, torch.nn.Module]) -> None:
         self.obj.update(modules)
 
 
-def load_pretrained_from_config(config_obj: LLMModelConfig, weights_save_path: Optional[str] = None) -> PreTrainedModel:
+def load_pretrained_from_config(
+    config_obj: LLMModelConfig,
+    model_config: Optional[AutoConfig] = None,
+    weights_save_path: Optional[str] = None,
+) -> PreTrainedModel:
     load_kwargs = {}
     if config_obj.quantization:
         # Apply quanitzation configuration at model load time
         load_kwargs["torch_dtype"] = getattr(torch, config_obj.quantization.bnb_4bit_compute_dtype)
         load_kwargs["quantization_config"] = config_obj.quantization.to_bitsandbytes()
 
+    if config_obj.model_parameters:
+        # Add any model specific parameters to the load kwargs
+        for param_name, param_value in config_obj.model_parameters.to_config().items():
+            # Not all parameters are supported by all models, so we only add the parameter to the load kwargs
+            # if it is supported by the model.
+            if hasattr(model_config, param_name):
+                load_kwargs[param_name] = param_value
+            else:
+                logger.warning(f"Parameter {param_name} is not supported by {config_obj.base_model}. Skipping.")
```

Comment:
It seems like we're going to log this warning even if the user didn't specify a value for this param (because we're not checking that the `param_value is None`). I would suggest changing the logic to something like:

```
if param_value is None:
  continue

if hasattr(model_config, param_name):
  load_kwargs[param_name] = param_value
else:
  logger.warning(f"Parameter {param_name} is not supported by {config_obj.base_model}. Skipping.")
```